File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        if len(nums) == 0:
            return []

        sum_nums = []
        sum_nums.append(nums[0])

        for i in range(1, len(nums)):
            sum_nums.append(nums[i]+sum_nums[i-1])
        
        return sum_nums

File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.java
class Solution {
    public int[] runningSum(int[] nums) {
        int[] sum_arr = new int[nums.length];

        sum_arr[0] = nums[0];

        for (int i = 1; i < nums.length; i++) {
            sum_arr[i] = sum_arr[i-1] + nums[i];
        }

        return sum_arr;
    }
}